### PR TITLE
Add swagger.yml as API definition

### PR DESCRIPTION
**Pull request by thinkh**
I commited an initial version of the api routes with parameters and return values. I also generated Flask code from this swagger file, but we cannot use it due to the Python 3.5 requirement (and Phovea Server is using Python 2.7). That's why I suggest to keep the swagger file as discussion and documentation for the API routes.

@gfrogat Please check the parameter and feel free to modify the swagger.yml. It would be good, if you can implement the API routes after the swagger.yml.

Tip: Use the [swagger editor](https://editor.swagger.io/) to view the file in a better format.

### CODE REVIEW

**Comment by thinkh** [comment]
We can do it the other way around and try to integrate and annotate the methods using the [Flask-RESTPlus](https://flask-restplus.readthedocs.io/en/stable/swagger.html) annotations, which generates a swagger test ui automatically from the code annotations.

File: malevo/swagger.yml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+          schema:
+            type: file
+          headers:
+            Content-type:
+              type: string
+              description: image/png; charset=utf-8
+        '400':
+          description: Invalid input value
+
+  /images/softmaxSprite:
```
Comment (gfrogat) [inline malevo/swagger.yml]: shouldn't a softmax sprite contain the softmax values for an image over several epochs? We're missing the epoch ids in the call

File: malevo/swagger.yml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          description: >-
+            Successful operation returns a list of image ids
+          schema:
+            type: array
+            items:
+              type: integer
+        '400':
+          description: Invalid input value
+
+  /class/crossEntropy:
```
Comment (gfrogat) [inline malevo/swagger.yml]: I would group crossEntropy under a separate tag `measures` as it's not really related to one class but specific for an img id and a specific epoch

File: malevo/swagger.yml [hunk 3]
```diff
@@ -0,0 +1,174 @@
+swagger: '2.0'
+info:
+  description: Malevo API
+  version: 0.1.0
+  title: Malevo API
+#host: localhost
+basePath: /api/malevo
+tags:
+  - name: class
```
Comment (gfrogat) [inline malevo/swagger.yml]: what do you think about renaming the class tag into confmat? The queries are now specific to one cell of a confusion matrix for an epoch.

We could then also extend the API later to give all the off-diagonal img ids for a specific ground truth or predicted label

File: malevo/swagger.yml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      tags:
+        - class
+      summary: Representative sample of images
+      description: >-
+        Returns a list of representative images for that matches all given class
+        ids
+      produces:
+        - application/json
+      parameters:
+        - name: dataset
```
Comment (gfrogat) [inline malevo/swagger.yml]: we have to be careful with the terminology here, dataset can refer to the actual training/test data and the run logs (what we are using here). Maybe just use `runId` to be more specific

File: malevo/swagger.yml [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+          in: query
+          description: Phovea dataset id
```
Comment (gfrogat) [inline malevo/swagger.yml]: I would avoid the mention of `Phovea` in the API specification.

File: malevo/swagger.yml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+          required: true
+          type: string
+        - name: epoch
+          in: query
+          description: Selected epoch id
+          required: true
+          type: integer
+        - name: groundTruthClassId
```
Comment (gfrogat) [inline malevo/swagger.yml]: do we need the `...ClassId` suffix in the call? Either add `...Id` to all names (e.g. epoch, dataset/run) or none
